### ISSUE TITLE
.Docker.CurrentContainerID is null in Github actions/workflows context

### ISSUE THREAD

**Issue by jonaseberle** [jwilder/docker-gen#315]
When running docker-gen in a container on Github actions the variable `.Docker.CurrentContainerID` is null and thus cannot be used in the template. 

I am not sure if this is only the case when running in the same container as the router.

I found https://github.com/jwilder/nginx-proxy/blob/master/nginx.tmpl and https://github.com/drud/ddev/blob/master/containers/ddev-router/nginx.tmpl being affected by this problem.

I tested with `runs-on: ubuntu-18.04` and an updated docker (`docker-ce amd64 5:19.03.5~3-0~ubuntu-bionic`). 

Do you have an idea what could be the reason?

I find the mounted /var/run/docker.sock inside the containers fully functional.

**Comment by lpeabody** [comment]
I did some more digging into this and Github Actions has this in /etc/docker/daemon.js:

```
{ "cgroup-parent": "/actions_job" }
```

And I wonder if that conflicts with:

https://github.com/jwilder/docker-gen/pull/158/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R156-R158

```
...
func GetCurrentContainerID() string {
	file, err := os.Open("/proc/self/cgroup")
...
```

**Comment by lpeabody** [comment]
Yup, that is 100% the problem. The Docker daemon has a default control group of `docker`. You can [optionally specify a different control group](https://docs.docker.com/engine/reference/commandline/dockerd/#miscellaneous-options), which the Github Actions engineers have opted to do, using `/actions_job` instead.

Below is the output of `docker exec docksal-vhost-proxy cat /proc/self/cgroup`:

```
docker exec docksal-vhost-proxy cat /proc/self/cgroup
12:rdma:/
11:cpu,cpuacct:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
10:devices:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
9:perf_event:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
8:memory:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
7:cpuset:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
6:hugetlb:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
5:blkio:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
4:freezer:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
3:net_cls,net_prio:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
2:pids:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
1:name=systemd:/actions_job/635245dfcea1b70ba4f1f1fe0ec1c439b24646e8239e0ebea4f2af2909d60930
0::/system.slice/containerd.service
```

The regex at https://github.com/jwilder/docker-gen/pull/158/files#diff-552f47512a00afe5fc6850cc9ddc830a6daeca162750e50aab4ed549685e0253R167 won't match that since `/docker` is not present.

So, definitely a bug with `GetCurrentContainerID`.

**Comment by buchdag** [comment]
@jonaseberle @lpeabody a fix is available there : https://github.com/buchdag/docker-gen/commit/8ca1084a608784c5d03692f20040a2984ed3922d

I'l trying to reach @jwilder to talk about this project maintenance, without success so far.

**Comment by lpeabody** [comment]
I just noticed it's been a couple of years. Thanks for trying to reach out, and very much appreciate being pointed in the direction of a fix for this.

Just in case it helps, I shall attempt to summon him as well: @jwilder we need your glorious maintainer power plz help :)

**Comment by lpeabody** [comment]
FYI I've successfully tested this fix on an Github Actions project. Thank you very much @buchdag.

**Comment by lmakarov** [comment]
Here's a quick and dirty hack we ended up doing:

```yaml
      # Hack to fix docker-gen in Github Actions
      # Drops docker daemon config with custom cgroup (which docker-gen does not support)
      # See https://github.com/jwilder/docker-gen/issues/315
      - name: Fix docker-gen in Github Actions
        run: |
          sudo rm /etc/docker/daemon.json
          sudo service docker restart
```

**Comment by lpeabody** [comment]
I think this is fixed in the latest release, 0.7.6.

**Comment by jonaseberle** [comment]
Thank you!
